module/zfs/dbuf.c

```diff
@@ -3984,6 +4044,153 @@ dbuf_sync_leaf_verify_bonus_dnode(dbuf_dirty_record_t *dr)
 #endif
 }
 
+static blkptr_t *
+dbuf_lightweight_bp(dbuf_dirty_record_t *dr)
+{
+	/* This must be a lightweight dirty record. */
+	ASSERT3P(dr->dr_dbuf, ==, NULL);
+	dnode_t *dn = dr->dr_dnode;
+
+	if (dn->dn_phys->dn_nlevels == 1) {
+		ASSERT3U(dr->dt.dll.dr_blkid, <, dn->dn_phys->dn_nblkptr);
+		return (&dn->dn_phys->dn_blkptr[dr->dt.dll.dr_blkid]);
+	} else {
+		dmu_buf_impl_t *parent_db = dr->dr_parent->dr_dbuf;
+		int epbs = dn->dn_indblkshift - SPA_BLKPTRSHIFT;
+		ASSERT3U(parent_db->db_level, ==, 1);
+		ASSERT3P(parent_db->db_dnode_handle->dnh_dnode, ==, dn);
+		ASSERT3U(dr->dt.dll.dr_blkid >> epbs, ==, parent_db->db_blkid);
```

Comment:
Does it make sense to convert some of these to VERIFYs?

Reply:
Sure, I'll change all of these.